src/Fixer/Basic/BracesFixer.php

```diff
@@ -178,892 +210,45 @@ protected function createConfigurationDefinition()
      */
-    private function findStatementEnd(Tokens $tokens, $parenthesisEndIndex)
-    {
-        $nextIndex = $tokens->getNextMeaningfulToken($parenthesisEndIndex);
-        $nextToken = $tokens[$nextIndex];
-
-        if (!$nextToken) {
-            return $parenthesisEndIndex;
-        }
-
-        if ($nextToken->equals('{')) {
-            return $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $nextIndex);
-        }
-
-        if ($nextToken->isGivenKind($this->getControlTokens())) {
-            $parenthesisEndIndex = $this->findParenthesisEnd($tokens, $nextIndex);
-
-            $endIndex = $this->findStatementEnd($tokens, $parenthesisEndIndex);
-
-            if ($nextToken->isGivenKind([T_IF, T_TRY, T_DO])) {
-                $openingTokenKind = $nextToken->getId();
-
-                while (true) {
-                    $nextIndex = $tokens->getNextMeaningfulToken($endIndex);
-                    $nextToken = isset($nextIndex) ? $tokens[$nextIndex] : null;
-                    if ($nextToken && $nextToken->isGivenKind($this->getControlContinuationTokensForOpeningToken($openingTokenKind))) {
-                        $parenthesisEndIndex = $this->findParenthesisEnd($tokens, $nextIndex);
-
-                        $endIndex = $this->findStatementEnd($tokens, $parenthesisEndIndex);
-
-                        if ($nextToken->isGivenKind($this->getFinalControlContinuationTokensForOpeningToken($openingTokenKind))) {
-                            return $endIndex;
-                        }
-                    } else {
-                        break;
-                    }
-                }
-            }
-
-            return $endIndex;
-        }
-
-        $index = $parenthesisEndIndex;
-
-        while (true) {
-            $token = $tokens[++$index];
-
-            // if there is some block in statement (eg lambda function) we need to skip it
-            if ($token->equals('{')) {
-                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
-
-                continue;
-            }
-
-            if ($token->equals(';')) {
-                return $index;
-            }
-
-            if ($token->isGivenKind(T_CLOSE_TAG)) {
-                return $tokens->getPrevNonWhitespace($index);
-            }
-        }
-
-        throw new \RuntimeException('Statement end not found.');
-    }
-
-    private function getControlTokens()
-    {
-        static $tokens = [
-            T_DECLARE,
-            T_DO,
-            T_ELSE,
-            T_ELSEIF,
-            T_FINALLY,
-            T_FOR,
-            T_FOREACH,
-            T_IF,
-            T_WHILE,
-            T_TRY,
-            T_CATCH,
-            T_SWITCH,
-        ];
-
-        return $tokens;
-    }
-
-    private function getControlContinuationTokens()
+    protected function createProxyFixers()
     {
-        static $tokens = [
-            T_CATCH,
-            T_ELSE,
-            T_ELSEIF,
-            T_FINALLY,
+        return [
```

Comment:
I wonder if we should add some integration test for relations between those newly created fixers...

Reply:
Maybe some priority tests indeed, otherwise I think tests in `BracesFixerTest` should be fine

Reply:
but on line v3, we gonna drop whole `BracesFixerTest` file, right?
so, we should have the replacement for it on v2.x as well. good we did not drop the cases in `BracesFixerTest` yet, but the new ones should be added as well

Reply:
I added some integration tests.